Quarkus 3.17.1+ does not like 'get' methods on Panache (Kotlin) Entities

### Describe the bug

I updated a project from 3.17.0 to 3.17.2 and started receiving exceptions in building and testing related to Hibernate bytecode enhancing:
```
    java.lang.RuntimeException at QuarkusTestExtension.java:628
        Caused by: java.lang.RuntimeException at AugmentActionImpl.java:355
            Caused by: io.quarkus.builder.BuildException at Execution.java:124
                Caused by: java.lang.reflect.UndeclaredThrowableException at ExtensionLoader.java:862
                    Caused by: java.util.concurrent.ExecutionException at FutureTask.java:122
                        Caused by: org.hibernate.bytecode.enhance.spi.EnhancementException at EnhancerImpl.java:131
                            Caused by: java.lang.StringIndexOutOfBoundsException at Preconditions.java:55
```

I traced it down to Hibernates Enhancer [1]. Starting from Quarkus 3.17.1 the Hibernate version used does not seem to like entities like
```kotlin
@Entity
class Entity(@Id var id: Long, var year: Int, var january: Double, var february: Double, ...) : PanacheEntityBase {
  operator fun get(month: Month) = ...

  companion object : PanacheCompanionBase<Entity, Long> {
   fun get(year: Int): Entity = ...
  }
}
``` 

The enhancer does not like both `get` methods. 


[1] https://github.com/hibernate/hibernate-orm/blob/e14ead0c0f0916cc8fdb8895e2cc481754a75583/hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java#L463

### Expected behavior

Quarkus builds the project without any issues.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

21

### Quarkus version or git rev

3.17.1+

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.11.1

### Additional information

I'm not sure whether this is a Quarkus or Hibernate issue; but since it's also occurring on Quarkus/ Panache companion objects, I decided to start here.

--- a/pom.xml
+++ b/pom.xml
@@ -71,7 +71,7 @@
         <jacoco.version>0.8.12</jacoco.version>
         <kubernetes-client.version>6.13.4</kubernetes-client.version> <!-- Please check with Java Operator SDK team before updating -->
         <rest-assured.version>5.5.0</rest-assured.version>
-        <hibernate-orm.version>6.6.3.Final</hibernate-orm.version> <!-- WARNING when updating, also align the versions below -->
+        <hibernate-orm.version>6.6.4.Final</hibernate-orm.version> <!-- WARNING when updating, also align the versions below -->
         <antlr.version>4.13.0</antlr.version> <!-- version controlled by Hibernate ORM's needs -->
         <bytebuddy.version>1.14.18</bytebuddy.version> <!-- version controlled by Hibernate ORM's needs -->
         <hibernate-commons-annotations.version>7.0.3.Final</hibernate-commons-annotations.version> <!-- version controlled by Hibernate ORM's needs -->

--- a/pom.xml
+++ b/pom.xml
@@ -75,7 +75,7 @@
         <antlr.version>4.13.0</antlr.version> <!-- version controlled by Hibernate ORM's needs -->
         <bytebuddy.version>1.14.18</bytebuddy.version> <!-- version controlled by Hibernate ORM's needs -->
         <hibernate-commons-annotations.version>7.0.3.Final</hibernate-commons-annotations.version> <!-- version controlled by Hibernate ORM's needs -->
-        <hibernate-reactive.version>2.4.2.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
+        <hibernate-reactive.version>2.4.3.Final</hibernate-reactive.version> <!-- highly sensitive to Hibernate ORM upgrades -->
         <hibernate-validator.version>8.0.2.Final</hibernate-validator.version>
         <hibernate-search.version>7.2.2.Final</hibernate-search.version>
 

